### PR TITLE
Add support for NFS Gateway

### PR DESCRIPTION
**Pull request by dang**
Add the Ganesha NFS gateway to the daemon.  This allows accessing the
RadosGW via NFS.

Signed-off-by: Daniel Gryniewicz <dang@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
A couple of things:

* can you also update the README?
* this: https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh#L917 needs to be update to
* can we have a 'demo' container version too?

Thanks!

**Comment by leseb** [comment]
Thanks for updating.
Code changed in the meantime because I just merged #352, I was referring to this line: https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh#L947

**Comment by dang** [comment]
Ah, right.  Fixed.

File: ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh [hunk 1]
```diff
@@ -913,8 +965,8 @@ case "$CEPH_DAEMON" in
   if [ ! -n "$CEPH_DAEMON" ]; then
     echo "ERROR- One of CEPH_DAEMON or a daemon parameter must be defined as the name "
     echo "of the daemon you want to deploy."
-    echo "Valid values for CEPH_DAEMON are MON, OSD, OSD_DIRECTORY, OSD_CEPH_DISK, OSD_CEPH_DISK_PREPARE, OSD_CEPH_DISK_ACTIVATE, OSD_CEPH_ACTIVATE_JOURNAL, MDS, RGW, RESTAPI, ZAP_DEVICE, RBD_MIRROR"
-    echo "Valid values for the daemon parameter are mon, osd, osd_directory, osd_ceph_disk, osd_ceph_disk_prepare, osd_ceph_disk_activate, osd_ceph_activate_journal, mds, rgw, restapi, zap_device, rbd_mirror"
+    echo "Valid values for CEPH_DAEMON are MON, OSD, OSD_DIRECTORY, OSD_CEPH_DISK, OSD_CEPH_DISK_PREPARE, OSD_CEPH_DISK_ACTIVATE, OSD_CEPH_ACTIVATE_JOURNAL, MDS, RGW, RGW_USER, RESTAPI, ZAP_DEVICE, RBD_MIRROR"
+    echo "Valid values for the daemon parameter are mon, osd, osd_directory, osd_ceph_disk, osd_ceph_disk_prepare, osd_ceph_disk_activate, osd_ceph_activate_journal, mds, rgw, rgw_user, restapi, zap_device, rbd_mirror"
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh]: and `nfs`?

**Comment by dang** [comment]
Sorry.  I saw RGW and read it as NFS...

**Comment by ktdreyer** [comment]
I don't see nfs-ganesha-rgw in EPEL 7 http://dl.fedoraproject.org/pub/epel/7/x86_64/n/ . Where does this package come from?

Why include nfs-ganesha-vfs ? Does the RGW FSAL depend on that?

**Comment by dang** [comment]
EPEL only includes the base FSALs for Ganesha, currently.  This is because it doesn't currently include Ceph or Gluster.

I included VFS in the package for testing purposes:  That way I (or a user) could test Ganesha independent of the Ceph cluster, and therefore verify each step separately.  VFS is not necessary for RGW or CEPH to work.

**Comment by ktdreyer** [comment]
If EPEL doesn't have the -rgw FSAL, does that mean this change will cause the yum operation to fail?

**Pull request merged**